packages/app/src/components/DownloadMenu/DownloadMenu.js

```diff
@@ -1,265 +1,55 @@
 import { VIS_TYPE_PIVOT_TABLE } from '@dhis2/analytics'
-import { useConfig, useDataEngine } from '@dhis2/app-runtime'
 import i18n from '@dhis2/d2-i18n'
-import {
-    FlyoutMenu,
-    Layer,
-    MenuSectionHeader,
-    MenuItem,
-    Popper,
-    IconImage24,
-    IconFileDocument24,
-    colors,
-} from '@dhis2/ui'
+import { FlyoutMenu, MenuSectionHeader } from '@dhis2/ui'
 import PropTypes from 'prop-types'
-import React, { useState, createRef } from 'react'
-import { connect } from 'react-redux'
-import { createSelector } from 'reselect'
-import {
-    apiDownloadImage,
-    apiDownloadData,
-    apiDownloadTable,
-} from '../../api/analytics.js'
-import { sGetChart } from '../../reducers/chart.js'
-import { sGetCurrent } from '../../reducers/current.js'
+import React from 'react'
+import { AdvancedSubMenu } from './AdvancedSubMenu.js'
 import {
-    sGetUiType,
-    sGetUiLayout,
-    sGetUiInterpretation,
-} from '../../reducers/ui.js'
-import MenuButton from '../MenuButton/MenuButton.js'
-
-const DenseMenuItem = ({ Icon, children, ...rest }) => (
-    <MenuItem dense icon={Icon && <Icon color={colors.grey600} />} {...rest}>
-        {children}
-    </MenuItem>
+    FILE_FORMAT_CSV,
+    FILE_FORMAT_XLS,
+    FILE_FORMAT_JSON,
+    FILE_FORMAT_XML,
+} from './constants.js'
+import { GraphicsMenu } from './GraphicsMenu.js'
+import { PlainDataSourceSubMenu } from './PlainDataSourceSubMenu.js'
+import { TableMenu } from './TableMenu.js'
+
+const DownloadMenu = ({ visType, downloadData, downloadImage }) => (
+    <FlyoutMenu>
+        {visType === VIS_TYPE_PIVOT_TABLE ? (
+            <TableMenu download={downloadData} />
+        ) : (
+            <GraphicsMenu download={downloadImage} />
+        )}
+        <MenuSectionHeader label={i18n.t('Plain data source')} />
+        <PlainDataSourceSubMenu
+            download={downloadData}
+            label={i18n.t('JSON')}
+            format={FILE_FORMAT_JSON}
+        />
+        <PlainDataSourceSubMenu
+            download={downloadData}
+            label={i18n.t('XML')}
+            format={FILE_FORMAT_XML}
+        />
+        <PlainDataSourceSubMenu
+            download={downloadData}
+            label={i18n.t('Microsoft Excel')}
+            format={FILE_FORMAT_XLS}
+        />
+        <PlainDataSourceSubMenu
+            download={downloadData}
+            label={i18n.t('CSV')}
+            format={FILE_FORMAT_CSV}
+        />
+        <AdvancedSubMenu download={downloadData} label={i18n.t('Advanced')} />
```

Comment:
It's not entirely clear to me that `download`, `downloadData` and `downloadImage` are actually functions. 
For the submenus (that now use have a `download` prop), these might as well just be called `onClick`, but a name with a prefix that indicates that its actually a function (like "on...", "perform...", "execute...", "start...", "do...") would work as well I suppose.
For the `useDownloadMenu` props, it might be more clear that these are actually functions when looking at the context (however just `downloadData` can be misinterpreted as "the data returned from a download" instead of "perform a download of data"), so maybe they can be left as is, but preferably they'd also be prefixed in a way so its clear that these are actually functions.

Reply:
Renamed props and functions.